src/speech_to_text_finetune/data_process.py

```diff
@@ -97,87 +122,188 @@ def load_dataset_from_dataset_id(
+
+    return _build_cv_dataset_from_df(
+        dataset_df=dataset_df,
+        audio_dir=audio_dir,
+        audio_clip_column=audio_clip_column,
+        is_spontaneous_speech=is_spontaneous_speech,
     )
-    common_voice = common_voice.select_columns(["audio", "sentence"])
 
-    return common_voice
+def _check_if_local_common_voice_is_spontaneous(cv_data_dir: Path) -> bool:
+    """
+    Check if the local Common Voice dataset is Spontaneous (SPS) or Scripted (SCS),
+    based on the expected directory structure and file names.
+    - SPS: contains `audios/` directory and `ss-corpus*.tsv` files
+    - SCS: contains `clips/` directory and `train.tsv`, `dev.tsv`, `test.tsv` files
+    """
+    entries = list(cv_data_dir.iterdir())
+    dir_names = {p.name for p in entries if p.is_dir()}
+    file_names = {p.name for p in entries if p.is_file()}
 
+    if "audios" in dir_names and any(name.startswith("ss-corpus") and name.endswith(".tsv") for name in file_names):
+        return True
+    elif "clips" in dir_names and {"train.tsv", "dev.tsv", "test.tsv"}.issubset(file_names):
+        return False
+    else:
+        raise ValueError("Unexpected dataset format. Could not determine if local Common Voice is SPS or SCS.")
 
 def _load_local_common_voice(cv_data_dir: str) -> DatasetDict:
     """
-    Load a local Common Voice dataset (as downloaded from the official Common Voice website) into a DatasetDict.
-    We only use the validated.tsv file to source the data to use for both training and testing.
+    Shared loader for local Common Voice (SPS/SCS).
+    Build a single DataFrame with `splits` for local Common Voice.
+    - SPS: scan `ss-corpus*.tsv` that already contains splits
+    - SCS: read `train.tsv`, `dev.tsv`, `test.tsv` and add a `splits` column
+    """
+    cv_data_dir = Path(cv_data_dir)
 
-    Args:
-        cv_data_dir (str): path to the local Common Voice dataset directory
+    is_spontaneous_speech = _check_if_local_common_voice_is_spontaneous(cv_data_dir)
+
+    if is_spontaneous_speech:
+        dataset_df = None
+        for file in cv_data_dir.iterdir():
+            if file.is_file() and file.name.startswith("ss-corpus") and file.name.endswith(".tsv"):
+                dataset_df = pd.read_csv(file, sep="\t")
+                break
+        if dataset_df is None:
+            raise FileNotFoundError("Could not find SPS `ss-corpus*.tsv` file.")
+        audio_dir = cv_data_dir / "audios"
+        audio_clip_column = "audio_file"
+    else:
+        train_df = pd.read_csv(cv_data_dir / "train.tsv", sep="\t")
+        train_df["split"] = "train"
+
+        dev_df = pd.read_csv(cv_data_dir / "dev.tsv", sep="\t")
+        dev_df["split"] = "dev"
+
+        test_df = pd.read_csv(cv_data_dir / "test.tsv", sep="\t")
+        test_df["split"] = "test"
+
+        dataset_df = pd.concat([train_df, dev_df, test_df], ignore_index=True)
+        audio_dir = cv_data_dir / "clips"
+        audio_clip_column = "path"
```

Comment:
I think these values, "clips/path" and "audios/audio_files", are used a couple of times in the code in different places (e.g. lines 166-170). I suggest defining them as globals at the top of this file (or elsewhere if you prefer)....also if for any reason in the future this needs to be updated it is a _tiny_ bit more straightforward